internal/v2/controller/http/common.go

```diff
@@ -55,3 +60,51 @@ func (c *V2HttpController) Metrics(writer http.ResponseWriter, request *http.Req
 	response := common.NewMetricsResponse(metrics)
 	c.sendResponse(writer, request, contractsV2.ApiMetricsRoute, response, http.StatusOK)
 }
+
+// Secret handles the request to add Device Service exclusive secret to the Secret Store
+// It returns a response as specified by the V2 API swagger in openapi/v2
+func (c *V2HttpController) Secret(writer http.ResponseWriter, request *http.Request) {
+	defer func() {
+		_ = request.Body.Close()
+	}()
+
+	provider := bootstrapContainer.SecretProviderFrom(c.dic.Get)
+	secretRequest := common.SecretRequest{}
+	err := json.NewDecoder(request.Body).Decode(&secretRequest)
+	if err != nil {
+		edgexError := errors.NewCommonEdgeX(errors.KindContractInvalid, "JSON decode failed", err)
+		c.sendEdgexError(writer, request, edgexError, sdkCommon.APIV2SecretRoute)
+		return
+	}
+
+	path, secrets := c.prepareSecret(secretRequest)
```

Comment:
nit -> secret?

Reply:
Hehe, I got the function name but missed the var name....